core/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java

```diff
@@ -785,4 +796,782 @@ private void writeAtomic(final String blobName, final BytesReference bytesRef) t
         }
     }
 
+    @Override
+    public void snapshot(IndexShard shard, SnapshotId snapshotId, IndexCommit snapshotIndexCommit, IndexShardSnapshotStatus snapshotStatus) {
```

Comment:
Is this implementation essentially unchanged when moved? I mean, modulo the obvious stuff.